apollo-router/src/plugins/include_query_plan.rs

```diff
@@ -0,0 +1,241 @@
+use futures::future::ready;
+use futures::stream::once;
+use futures::StreamExt;
+use http::HeaderValue;
+use serde_json_bytes::json;
+use tower::util::BoxService;
+use tower::BoxError;
+use tower::ServiceExt as TowerServiceExt;
+
+use crate::layers::ServiceExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::register_plugin;
+use crate::services::QueryPlannerContent;
+use crate::services::QueryPlannerRequest;
+use crate::services::QueryPlannerResponse;
+use crate::services::RouterRequest;
+use crate::services::RouterResponse;
+
+const INCLUDE_QUERY_PLAN_HEADER_NAME: &str = "X-Apollo-Query-Plan";
+const INCLUDE_QUERY_PLAN_ENV: &str = "APOLLO_INCLUDE_QUERY_PLAN";
+const QUERY_PLAN_CONTEXT_KEY: &str = "experimental::include_query_plan.plan";
+const ENABLED_CONTEXT_KEY: &str = "experimental::include_query_plan.enabled";
```

Comment:
Similar justification as above, but what about `expose_query_plan`?

I'm not sure if that's intuitive or not though.